src/backend/drm/output.rs

```diff
@@ -0,0 +1,459 @@
+use std::{
+    cell::RefCell,
+    collections::HashMap,
+    os::fd::AsFd,
+    sync::{Arc, RwLock},
+};
+
+use drm::control::{self, connector, crtc, Mode};
+use drm_fourcc::{DrmFormat, DrmFourcc, DrmModifier};
+
+use crate::{
+    backend::{
+        allocator::{
+            dmabuf::{AsDmabuf, Dmabuf},
+            gbm::GbmDevice,
+            Allocator,
+        },
+        renderer::{element::RenderElement, Bind, Color32F, DebugFlags, Renderer, Texture},
+    },
+    output::OutputModeSource,
+};
+
+use super::{
+    compositor::{
+        DrmCompositor, FrameError, FrameMode, FrameResult, RenderFrameError, RenderFrameErrorType,
+        RenderFrameResult,
+    },
+    exporter::ExportFramebuffer,
+    DrmDevice, DrmError, Planes,
+};
+
+type CompositorList<A, F, U, G> = Arc<RwLock<HashMap<crtc::Handle, RefCell<DrmCompositor<A, F, U, G>>>>>;
```

Comment:
If i understand correctly, `RefCell` is `!Sync`, which would make `CompositorList` both `!Sync` and `!Send`, making `Arc<RwLock<>>` pointless.

Reply:
Indeed, sigh. `!Sync` is intended/expected, it should be `Send` , I guess this is blocked on the `PixmanRenderer` not being `Send` :(

Reply:
Sorry I'm a little confused. `Arc<T>`  is `Sync` or `Send` requires `T` to be both `Sync` and `Send`, but `RefCell<T>` is unconditional `!Sync`. I think `RefCell<T>` should be removed here.

Reply:
`RwLock` or `Mutex` add the `Sync` bound, so our `T` is just required to be `Send`, which is indeed blocked on pixman-rs

Reply:
Looking at it again we indeed need to use Mutex instead of the Refcell because RwLock requires Sync for read access.

This is the easy part, unfortunately DrmCompositor is not Send atm. Currently looking into this.

Reply:
Okay, `DrmCompositor` is now `Send` as long as the allocator buffer and the framebuffer are `Send + Sync`. This allows `DrmOutputManager` to be `Send` and `DrmOutput` to implement `Send` and `Sync`.

## Notes
- [ ] This currently needs `unsafe impl Sync` for `GbmBuffer` which should be removed when [this](https://github.com/Smithay/gbm.rs/pull/45) lands
- [ ] The unsafe `Sync` impl on `PixmanRenderer` should be fine atm, but we need to address this in [`pixman-rs`](https://github.com/cmeissl/pixman-rs/pull/18)

Reply:
gbm 0.16.1 published